rclcpp/test/rclcpp/test_init_options.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <vector>
+
+#include "rcl/allocator.h"
+#include "rcl/domain_id.h"
+
+#include "rclcpp/init_options.hpp"
+
+
+TEST(TestInitOptions, test_construction) {
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  auto options = rclcpp::InitOptions(allocator);
+  const rcl_init_options_t * rcl_options = options.get_rcl_init_options();
+  ASSERT_TRUE(rcl_options != nullptr);
+  ASSERT_TRUE(rcl_options->impl != nullptr);
+
+  {
+    auto options_copy = rclcpp::InitOptions(options);
+    const rcl_init_options_t * rcl_options_copy = options_copy.get_rcl_init_options();
+    ASSERT_TRUE(rcl_options_copy != nullptr);
+    ASSERT_TRUE(rcl_options_copy->impl != nullptr);
+  }
+
+  {
+    auto options_copy = options;
+    const rcl_init_options_t * rcl_options_copy = options_copy.get_rcl_init_options();
+    ASSERT_TRUE(rcl_options_copy != nullptr);
+    ASSERT_TRUE(rcl_options_copy->impl != nullptr);
+  }
+}
+
+TEST(TestInitOptions, test_initialize_logging) {
+  {
+    auto options = rclcpp::InitOptions();
+    EXPECT_TRUE(options.auto_initialize_logging());
+    const rcl_init_options_t * rcl_options = options.get_rcl_init_options();
+    ASSERT_TRUE(rcl_options != nullptr);
+    ASSERT_TRUE(rcl_options->impl != nullptr);
```

Comment:
I don't understand why this three lines have to be repeated here.
This was already tested in the construction test, and it's not related to the `auto_initialize_logging` API.

Reply:
right, i was thinking that `just in case`, but as you mentioned, It'd be better to keep the consistency.